api/src/test/java/org/pmiops/workbench/firecloud/FireCloudServiceImplTest.java

```diff
@@ -188,4 +190,29 @@ public void testCreateAllOfUsBillingProject() throws Exception {
     assertThat(request.isHighSecurityNetwork()).isTrue();
     assertThat(request.getServicePerimeter()).isEqualTo(servicePerimeter);
   }
+
+  @Test
+  public void testCreateAllOfUsBillingProject_v2BillingApi() throws Exception {
+    final String servicePerimeter = "a-cloud-with-a-fence-around-it";
+    // confirm that this value is no longer how we choose perimeters
+    workbenchConfig.firecloud.vpcServicePerimeterName = "something else";
+    workbenchConfig.featureFlags.enableFireCloudV2Billing = true;
+
+    service.createAllOfUsBillingProject("project-name", servicePerimeter);
+
+    ArgumentCaptor<FirecloudCreateRawlsBillingProjectFullRequest> captor =
+        ArgumentCaptor.forClass(FirecloudCreateRawlsBillingProjectFullRequest.class);
+    verify(billingV2Api).createBillingProjectFullV2(captor.capture());
+    FirecloudCreateRawlsBillingProjectFullRequest request = captor.getValue();
+
+    // N.B. FireCloudServiceImpl doesn't add the project prefix; this is done by callers such
+    // as BillingProjectBufferService.
+    assertThat(request.getProjectName()).isEqualTo("project-name");
+    // FireCloudServiceImpl always adds the "billingAccounts/" prefix to the billing account
+    // from config.
+    assertThat(request.getBillingAccount()).isEqualTo("billingAccounts/test-billing-account");
+    assertThat(request.isEnableFlowLogs()).isTrue();
+    assertThat(request.isHighSecurityNetwork()).isTrue();
+    assertThat(request.getServicePerimeter()).isEqualTo(servicePerimeter);
```

Comment:
i think this test might be asserting too many things? It looks like all of these fields will be the same regardless of whether billing v1 or v2 is being used. The verify assertion on line 205 is the only line that really checks the new behavior, I think the rest of these will pass even if we disable the feature flag.

I think it's fine to leave 1 or 2 in here as a sanity check but I think most of it can just be deleted, especially given that the same code paths are being tested in the test above.

Reply:
right. yeah, that makes sense.